Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -40,16 +40,21 @@ static InputStream channelPositionInputStream(SeekableChannelsProvider provider,
     /**
      * Create a new {@link SeekableChannelContext} object for creating read channels via this provider.
      */
-    SeekableChannelContext makeContext();
+    SeekableChannelContext makeReadContext();
 
     /**
      * Create a new "single-use" {@link SeekableChannelContext} object for creating read channels via this provider.
      * This is meant for contexts that have a short lifecycle and expect to read a small amount from a read channel.
      */
-    default SeekableChannelContext makeSingleUseContext() {
-        return makeContext();
+    default SeekableChannelContext makeSingleUseReadContext() {
+        return makeReadContext();
     }
 
+    /**
+     * Create a new {@link SeekableChannelContext} object for creating output streams via this provider.
+     */
+    SeekableChannelContext makeWriteContext();
```

Comment:
I think we should use a new, barebones interface for this. Having it be Safecloseable is fine, but I don't think we want / need the other read-based stuff.